packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -731,80 +734,90 @@ class FlutterPlugin implements Plugin<Project> {
         }
     }
 
-    // Add a task that can be called on Flutter projects that prints application id of a build
-    // variant.
+    // Add a task that can be called on Flutter projects that dumps app link related project
+    // settings into a json file.
     //
-    // This task prints the application id in this format:
+    // See https://developer.android.com/training/app-links/ for more information about app link.
     //
-    // ApplicationId: com.example.my_id
+    // The json will be stored in <project>/build/app/app-link-settings-<variant>.json
     //
-    // Format of the output of this task is used by `AndroidProject.getApplicationIdForVariant`.
-    private static void addTasksForPrintApplicationId(Project project) {
-        project.android.applicationVariants.all { variant ->
-            // Warning: The name of this task is used by `AndroidProject.getApplicationIdForVariant`.
-            project.tasks.register("print${variant.name.capitalize()}ApplicationId") {
-                description "Prints out application id for the given build variant of this Android project"
-                doLast {
-                    println "ApplicationId: ${variant.applicationId}";
-                }
-            }
-        }
-    }
-
-    // Add a task that can be called on Flutter projects that prints app link domains of a build
-    // variant.
-    //
-    // The app link domains refer to the host attributes of data tags in the apps' intent filters
-    // that support http/https schemes. See
-    // https://developer.android.com/guide/topics/manifest/intent-filter-element.
-    //
-    // This task prints app link domains in this format:
+    // An example json:
+    // {
+    //   applicationId: "com.example.app",
+    //   deeplinks: [
+    //     {"scheme":"http", "host":"example.com", "path":".*"},
+    //     {"scheme":"https","host":"example.com","path":".*"}
+    //   ]
+    // }
     //
-    // Domain: domain.com
-    // Domain: another-domain.dev
-    //
-    // Format of the output of this task is used by `AndroidProject.getAppLinkDomainsForVariant`.
-    private static void addTasksForPrintAppLinkDomains(Project project) {
+    // The output file is parsed and used by devtool.
+    private static void addTasksForDumpAppLinkSettings(Project project) {
         project.android.applicationVariants.all { variant ->
-            // Warning: The name of this task is used by `AndroidProject.getAppLinkDomainsForVariant`.
-            project.tasks.register("print${variant.name.capitalize()}AppLinkDomains") {
-                description "Prints out app links domain for the given build variant of this Android project"
+            // Warning: The name of this task is used by AndroidBuilder.dumpsAppLinkSettings
+            project.tasks.register("dump${variant.name.capitalize()}AppLinkSettings") {
```

Comment:
I recognize names are hard but I am not sure "dump" helps clarify what this task does.  Think something like format or document or record would probably be more descriptive. 

Additionally we have had a stable release since print<variant name>AppLinkDomains was added so this is technically a breaking change. I am open to the argument that no one probably depended on it but wanted to call that out.

Reply:
Will try to come up with a better name

As per our breaking change rule, Only customer test failures counts as breaking change; otherwise, it would be us to determine whether we should send out a courtesy email. 

Like you said, I doubt anyone is using these gradle tasks and Android Project API. So I think we should be fine